test/src/main/java/org/apache/accumulo/test/functional/ConcurrentTableNameOperationsIT.java

```diff
@@ -0,0 +1,414 @@
+      newNamespaces.removeAll(namespacesBefore);
+      assertEquals(Set.of(namespaceName), newNamespaces,
+          "Expected exactly one new namespace: " + namespaceName);
+
+      client.namespaceOperations().delete(namespaceName);
+    }
+
+    pool.shutdown();
+  }
+
+  /**
+   * Test that when several threads attempt to rename to the same namespace name simultaneously,
+   * only one rename succeeds.
+   */
+  @Test
+  public void renameNamespace() throws Exception {
+    final int numTasks = 10;
+    final int numIterations = 3;
+    ExecutorService pool = Executors.newFixedThreadPool(numTasks);
+    String[] targetNamespaceNames = getUniqueNames(numIterations);
+
+    for (String targetNamespaceName : targetNamespaceNames) {
+      // multiple source namespaces for rename ops
+      List<String> sourceNamespaces = new ArrayList<>();
+      for (int i = 0; i < numTasks; i++) {
+        String sourceNamespace = targetNamespaceName + "_source_" + i;
+        client.namespaceOperations().create(sourceNamespace);
+        sourceNamespaces.add(sourceNamespace);
+      }
+
+      Set<String> namespacesBefore = client.namespaceOperations().list();
+
+      int successCount = runConcurrentNamespaceOperation(pool, numTasks, (index) -> {
+        client.namespaceOperations().rename(sourceNamespaces.get(index), targetNamespaceName);
+        return true;
+      });
+
+      assertEquals(1, successCount, "Expected only one rename operation to succeed");
+      assertTrue(client.namespaceOperations().exists(targetNamespaceName),
+          "Expected target namespace " + targetNamespaceName + " to exist");
+
+      Set<String> namespacesAfter = client.namespaceOperations().list();
+      assertEquals(namespacesBefore.size(), namespacesAfter.size(),
+          "Expected same namespace count (rename operation)");
+      assertTrue(namespacesAfter.contains(targetNamespaceName),
+          "Expected target namespace in final list");
+
+      for (String sourceNamespace : sourceNamespaces) {
+        if (client.namespaceOperations().exists(sourceNamespace)) {
+          client.namespaceOperations().delete(sourceNamespace);
+        }
+      }
+      if (client.namespaceOperations().exists(targetNamespaceName)) {
+        client.namespaceOperations().delete(targetNamespaceName);
+      }
+    }
+
+    pool.shutdown();
+  }
+
+  private int runConcurrentTableOperation(ExecutorService pool, int numTasks,
+      ConcurrentOperation operation) throws ExecutionException, InterruptedException {
+    return runConcurrentOperation(pool, numTasks, operation, TableExistsException.class);
+  }
+
+  private int runConcurrentNamespaceOperation(ExecutorService pool, int numTasks,
+      ConcurrentOperation operation) throws ExecutionException, InterruptedException {
+    return runConcurrentOperation(pool, numTasks, operation, NamespaceExistsException.class);
+  }
+
+  private int runConcurrentOperation(ExecutorService pool, int numTasks,
```

Comment:
This method and the functional interface avoids a lot of boiler plate code in the test.